### PR TITLE
Fixing Common DeepChem errors

### LINKED ISSUE #2513: DeepChem Common Errors
When using DeepChem, there are some classes of common errors I've observed that don't have great error messages. I'm starting this thread up for us to start documenting some patterns which hopefully can lead to improved error messages:

- **Using the wrong featurization:** When copy/pasting example code, you'll often do things like using ECFP featurizations with graph convs which will yield a very confusing TensorFlow error. This ideally should have a natural error message suggesting the user check their featurization.
- **Padding:** The generators in KerasModel pad by default. When using custom models, this often confuses newcomers since they enter in a dataset of say 100 elements with batch size 64 and are very surprised to receive back 128 predictions. Some warning should be emitted along with a suggested fix

I'll add on more errors here as I come across them. Please add on your own observed confusing DeepChem behavior!

### PR DESCRIPTION
**Pull request by manangoel99**
This pull request tackles some common deep chem errors to improve the user experience,

Fixes #2513 

- [x] Raising an informative runtime error when incorrect featurization is used with ```GraphConv``` model by adding a try/catch block in ```deepchem/feat/mol_graphs.py```. Added a corresponding unit test in ```deepchem/models/tests/test_graph_models.py```
- [x] Update the tutorials according to the latest versions of deepchem and other supporting modules.

I will add more bug fixes to the same pull request mentioned in #2513.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.22.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by manangoel99** [comment]
This is my first contribution to DeepChem. Please give it a review and let me know how I can improve!

File: deepchem/feat/mol_graphs.py [hunk 1]
```diff
@@ -264,8 +264,11 @@ def agglomerate_mols(mols, max_deg=10, min_deg=0):
     num_mols = len(mols)
 
     # Combine the features, then sort them by (atom_degree, mol_index)
-    atoms_by_deg = np.concatenate([x.atom_features for x in mols])
-    degree_vector = np.concatenate([x.degree_list for x in mols], axis=0)
+    try:
+      atoms_by_deg = np.concatenate([x.atom_features for x in mols])
+      degree_vector = np.concatenate([x.degree_list for x in mols], axis=0)
+    except:
+      raise RuntimeError('Incorrect featurization used for Convolution model')
```
Comment (rbharath) [inline deepchem/feat/mol_graphs.py:271]: I like the try/catch statement approach, but we need to be a bit careful here. We don't want to catch all exceptions so we should make sure to catch the precise exception that is raised if ECFP is used with GraphConv (or another featurization)

**Review by rbharath — COMMENTED** [review]
We need to tighten up the try/catch statement here but otherwise this is a good start! Thanks for the PR and sorry for the very late review :)

File: deepchem/feat/mol_graphs.py (hunk 1 above)
Comment (manangoel99) [reply deepchem/feat/mol_graphs.py:271]: That's what my initial thought was as well but obtaining the dataset object and getting the model object are separate and the only time they come together is when the ```model.fit``` function is called.
This was the place where I found that the behaviours diverged depending on featurization type.

Comment (rbharath) [reply deepchem/feat/mol_graphs.py:271]: One alternative might be to check that the entries in the incoming dataset are `ConvMol` objects (perhaps just checking the first object for speed) and raise an error if so

Comment (manangoel99) [reply deepchem/feat/mol_graphs.py:271]: Yeah I think that would be much faster but again that would happen in the `model.fit` function if I understand correctly.

File: deepchem/feat/mol_graphs.py [hunk 2]
```diff
@@ -263,6 +263,11 @@ def agglomerate_mols(mols, max_deg=10, min_deg=0):
 
     num_mols = len(mols)
 
+    # Check if the all the objects in the mols array are ConvMol objects
+    for m in mols:
+      if type(m) != ConvMol:
+        raise RuntimeError('Incorrect featurization used for Convolution model')
```
Comment (rbharath) [inline deepchem/feat/mol_graphs.py:269]: I think this should likely be a `ValueError` since it's for an incorrect argument type

**Review by rbharath — COMMENTED** [review]
The design on the `ConvMol` error catch looks better now! We need to only make some minor fixes there.

Can you split the tutorial changes to separate PRs? I'd recommend fixing 1/2 tutorials per PR to keep the review manageable.

File: deepchem/feat/mol_graphs.py (hunk 2 above)
Comment (manangoel99) [reply deepchem/feat/mol_graphs.py:269]: Sure! I'll change the error type and create a new couple of PRs for the tutorials.

**Comment by rbharath** [comment]
Closing this old PR for cleanup
